internal/datasource/config/config_writer.go

```diff
@@ -36,46 +37,40 @@ type (
 		Complete() (err error)
 	}
 
-	Client struct {
-		Timeout time.Duration
-	}
-
-	ConfigWriterParameters struct {
-		configClient client.HTTPConfigClientInterface
-		Client       Client
-		cachePath    string
-	}
-
 	ConfigWriter struct {
 		configClient       client.HTTPConfigClientInterface
 		previouseFileCache FileCache
 		currentFileCache   FileCache
 		cachePath          string
 		dataplaneConfig    config.DataplaneConfig
+		allowedDirectories []string
 	}
 
 	// map of files with filepath as key
 	FileCache = map[string]*instances.File
 )
 
-func NewConfigWriter(configWriterParameters *ConfigWriterParameters, instanceID string) *ConfigWriter {
-	if configWriterParameters.cachePath == "" {
-		configWriterParameters.cachePath = fmt.Sprintf(cacheLocation, instanceID)
+func NewConfigWriter(configClient client.HTTPConfigClientInterface, cachePath string,
```

Comment:
Might be out of scope of this PR but we probably should rename `HTTPConfigClientInterface` to `ConfigClient` to make it more generic

Reply:
I can make the change in this PR, I don't mind